modules/navier_stokes/test/src/postprocessors/TestFaceToCellReconstruction.C

```diff
@@ -0,0 +1,71 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "TestFaceToCellReconstruction.h"
+
+// MOOSE includes
+#include "MooseMesh.h"
+
+registerMooseObject("NavierStokesTestApp", TestFaceToCellReconstruction);
+
+InputParameters
+TestFaceToCellReconstruction::validParams()
+{
+  InputParameters params = GeneralPostprocessor::validParams();
+  params.addClassDescription("Computes reconstruction error for face centered functor.");
+  return params;
+}
+
+TestFaceToCellReconstruction::TestFaceToCellReconstruction(const InputParameters & parameters)
+  : GeneralPostprocessor(parameters),
+    _reconstruction_error(0.0),
+    _face_values(_subproblem.mesh(), _subproblem.mesh().meshSubdomains(), "face_values")
+{
+}
+
+void
+TestFaceToCellReconstruction::initialize()
+{
+  for (auto & fi : _fe_problem.mesh().faceInfo())
+  {
+    const auto & face_center = fi->faceCentroid();
+    RealVectorValue face_value(
+        -sin(face_center(0)) * cos(face_center(1)), cos(face_center(0)) * sin(face_center(1)), 0);
+    _face_values[fi->id()] = face_value;
+  }
+
+  // Loop over the local elements if we have distributed tests
+  auto begin = _fe_problem.mesh().getMesh().active_local_elements_begin();
+  auto end = _fe_problem.mesh().getMesh().active_local_elements_end();
+
+  for (const Elem * elem : as_range(begin, end))
+  {
+    const auto elem_arg = makeElemArg(elem);
+    const Point & elem_centroid = _fe_problem.mesh().elemInfo(elem->id()).centroid();
+    const Real elem_volume = _fe_problem.mesh().elemInfo(elem->id()).volume();
```

Comment:
This won't be the correct volume in non-XYZ  coordinate systems but for a test object this is fine

Reply:
Actually might as well:
```suggestion
    const Real elem_volume = _assembly.elementVolume(elem);
```
And now I'm realizing that `ElemInfo::volume()` returns wrong values for non-XYZ coordinate systems. This is undoubtedly a hold-over from Robert's work in FaceInfo ... this definitely troubles me. We should make an issue and probably remove this API because it's very dangerous

Reply:
I've opened https://github.com/idaholab/moose/issues/23094